src/Components/CreateImageWizardV2/steps/FileSystem/FileSystemConfiguration.tsx

```diff
@@ -28,22 +30,76 @@ import {
   selectImageTypes,
   removePartition,
   selectPartitions,
+  changeIsNextButtonPristine,
+  selectIsNextButtonPristine,
+  selectFileSystemPartitionMode,
 } from '../../../../store/wizardSlice';
 import UsrSubDirectoriesDisabled from '../../UsrSubDirectoriesDisabled';
 import { ValidatedTextInput } from '../../ValidatedTextInput';
+import {
+  getDuplicateMountPoints,
+  isFileSystemConfigValid,
+} from '../../validators';
 
 export type Partition = {
   id: string;
   mountpoint: string;
   min_size: string;
 };
+type FileSystemConfigurationStepFooterPropTypes = {
+  setNextDisable(isNextDisable: boolean): void;
+  isNextDisable: boolean;
+};
+
+export const FileSystemStepFooter: React.FunctionComponent<
+  FileSystemConfigurationStepFooterPropTypes
+> = ({ setNextDisable, isNextDisable }) => {
+  const { goToNextStep, goToPrevStep, close } = useWizardContext();
+  let isValid = false;
+  const dispatch = useAppDispatch();
+  const fileSystemPartitionMode = useAppSelector((state) =>
+    selectFileSystemPartitionMode(state)
+  );
+  const partitions = useAppSelector((state) => selectPartitions(state));
+  if (
+    fileSystemPartitionMode === 'automatic' ||
+    isFileSystemConfigValid(partitions)
+  ) {
+    isValid = true;
+  } else isValid = false;
+
+  const onValidate = () => {
+    dispatch(changeIsNextButtonPristine());
+    if (!isValid) {
+      setNextDisable(true);
+    } else {
+      goToNextStep();
+    }
+  };
```

Comment:
The pristinity (is that a word? :sweat_smile:) of the button might need to be reset at some point, which is connected to my other comment in wizardSlice file.

Try going to FSC step, creating a duplicate mount point and clicking Next. Everything renders correctly and `isNextButtonPristine` is set to false. So far so good.

Resolve the duplicity, the Next button should be enabled again.

But now if you again add a duplicate mount point, the error is rendered immediately and the Next button stays enabled until it is clicked.

![image](https://github.com/osbuild/image-builder-frontend/assets/49452678/673e146b-d283-4710-947d-8031345e909b)

Can we reset the pristinity of the Next button after the duplicity was resolved somehow? If that's too complicated, could we at least disable the Next button when the alert is rendered?

Reply:
I agree, let's aim for 1:1 behavior with what we have in prod for now.